integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/testsupport/util/LuceneTckBackendFeatures.java

```diff
@@ -39,4 +39,29 @@ public boolean projectionPreservesEmptySingleValuedObject(ObjectStructure struct
 	public boolean reliesOnNestedDocumentsForMultiValuedObjectProjection() {
 		return true;
 	}
+
+	public boolean supportsHighlighterUnifiedTypeNoMatchSize() {
+		// Lucene default unified highlighter does not support no-match-size setting.
+		// While in ES a custom highlighter is used that allows for such option.
+		return false;
```

Comment:
`supportsHighlighterUnifiedTypeNoMatchSize` + makes sense
`supportsHighlighterUnifiedTypeMultipleFragmentsAsSeparateItems` - this shouldn't be an exception since if we do so we can just remove support for the Unified highlighter in the Lucene backend.
`supportsHighlighterUnifiedTypeFragmentSize` + makes sense 
`supportsHighlighterUnifiedTypeMaxAnalyzedOffsetOnFieldsWithTermVector` + makes sense

Reply:
`supportsHighlighterUnifiedTypeNoMatchSize` - only logs a warning, since this setting partially works. Meaning if `no-match-size > 0` then (all) text will be returned, but if the value is not set (`no-match-size == 0`) then an empty list is returned. Hence only a warning and letting the user know if they set the value to maxint then the warning will disappear. 

`supportsHighlighterUnifiedTypeFragmentSize` - will throw an exception on builder method. So user will know right away when creating a query.

`supportsHighlighterUnifiedTypeMaxAnalyzedOffsetOnFieldsWithTermVector` - since we don't know to which filed the config will apply ... the check happens on `request(..)` and if needed an exception is thrown.